### ISSUE TITLE
Missing cursor theme in GNOME

### ISSUE THREAD

**Issue by hyperfekt** [NixOS/nixpkgs#140505]
### Describe the bug
An ugly cursor is displayed that turns into a large white square on certain surfaces.

### Steps To Reproduce
Change config to the following:
```
services.xserver = {
  enable = true;
  displayManager.gdm.enable = true;
  desktopManager.gnome.enable = true;
};
```

### Expected behavior
A modern cursor theme that ships with GNOME by default is applied.

### Additional context
The journal displays `gnome-shell-wr[2198]: No cursor theme available, please install a cursor theme`

### Notify maintainers
@jtojnar

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.19, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.11pre320334.82155ff501c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - gnome.gnome-shell
# a list of nixos modules affected by the problem
module:
 - services/x11/desktop-managers/gnome.nix
```

**Comment by jtojnar** [comment]
I have not seen anything like that, what surfaces did you see it one?

**Comment by hyperfekt** [comment]
It's everywhere on the Activities screen, or for example when hovering over the bar at the top of the screen. If you're on the latest unstable there must be another factor that seems unrelated at first but is causing it. Presumably you also aren't getting the failure to find a cursor theme in your log?

**Comment by jtojnar** [comment]
Yeah, I maintain GNOME on unstable and that is also the only channel where I support it.

I do not see the message in the journal:

```
$ journalctl -xa --since '-1 week' | grep 'cursor theme available'
$ 
```

Though I usually do not reboot my system after rebuild so I might not be running the latest Shell.

But I am not aware of any recent change that could affect this.

Did this only start recently for you?

Do you run X or Wayland session?

**Comment by hyperfekt** [comment]
I'd only used KDE before very recently so I can't compare it to older nixpkgs revs. COuld there be impurities from KDE that are causing this? I tried to make sure I didn't have any options left from it.
The cursor theme is still missing with `wayland = false`, but the white square disappears.
Here's an image that should illustrate that the Adwaita theme (which to my understanding is the intended default) is not being applied. This is what shows when hovering over a link.

<img src="https://user-images.githubusercontent.com/9434715/135907374-02d7d1bd-ebf9-4468-9f1c-b4624b1d23a5.png">

**Comment by jtojnar** [comment]
I have heard about Plasma state messing up with systems long after it was gone.

I would look at what is stored in the settings:

```
$ dconf read /org/gnome/desktop/interface/cursor-theme
'Adwaita'
```

Maybe also look at `~/.config/gtk-3.0/settings.ini`.

**Comment by hyperfekt** [comment]
That's it! It was set to `breeze_cursors`. Not sure how we would go about preventing that without applying the setting with an overlay database (as proposed in #54150).

**Comment by jtojnar** [comment]
I would probably also expect GNOME to fall back to Adwaita if the selected cursor theme does not exist.

**Comment by ryantm** [comment]
I have a similar issue when I am trying to switch from Gnome X to Gnome Wayland:

```
[ryantm@home2:~]$ dconf read /org/gnome/desktop/interface/cursor-theme

[ryantm@home2:~]$ gnome-terminal
# Failed to load cursor theme Adwaita
# Failed to load cursor theme Adwaita
# Error creating terminal: Message recipient disconnected from message bus without replying
```

**Comment by ryantm** [comment]
When I deleted `~/.config/dconf` it resolved itself. But that also got rid of all my customization, so probably not a recommended workaround.

**Comment by veluria** [comment]
> When I deleted `~/.config/dconf` it resolved itself. But that also got rid of all my customization, so probably not a recommended workaround.

Thanks. I had to resort to this as well. Off to a fresh start, I guess.

**Comment by l0b0** [comment]
> That's it! It was set to `breeze_cursors`.

Same. Fixing it was as simple as `dconf reset /org/gnome/desktop/interface/cursor-theme`.

**Comment by TobTobXX** [comment]
My icons were also messed up after switching from Plasma. This fixed it:
```
dconf reset /org/gnome/desktop/interface/icon-theme
```

(Just in case anyone has the same error.)
